.github/workflows/deploy-demo.yml

```diff
@@ -1,41 +1,83 @@
+  # Runs on pushes targeting the default branch
   push:
-    branches:
-      - demo  
+    branches: ["demo"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+env:
+  PYTHON_VERSION: '3.9'
+
 jobs:
-  deploy_demo:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
     runs-on: ubuntu-latest
-    env:
-      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v4
-
-    - name: Setup environment
-      run: |-
-        cd demo-project
-        echo "AWS_ECR_URL=public.ecr.aws/g0x0s3o2/kedro-viz-live-demo" >> $GITHUB_ENV
-        echo "KEDRO_VIZ_VERSION=$(cat .version)" >> $GITHUB_ENV
-
-    - name: Install AWS CLI
-      run: pip3 install awscli
-
-    - name: Build demo container image
-      run: |-
-        cd demo-project
-        echo "kedro_viz==$KEDRO_VIZ_VERSION" >> src/docker_requirements.txt
-        docker build -t $AWS_ECR_URL:$KEDRO_VIZ_VERSION .
-        aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
-        docker push $AWS_ECR_URL:$KEDRO_VIZ_VERSION
-
-    - name: Create a new lightsail deployment
-      run: |-
-        cd demo-project
-        # run https://docs.aws.amazon.com/cli/latest/reference/lightsail/create-container-service-deployment.html#create-container-service-deployment
-        aws lightsail create-container-service-deployment --region eu-west-2 --cli-input-json file://./lightsail.json
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}    
+
+      - name: Cache python packages for Linux
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ubuntu-latest-python-3.9            
+
+      - name: Install Kedro and other Python Dependencies
+        uses: "./.github/actions/install_kedro_and_python_dependencies"
+
+      - name: Setup Node.js and Install Dependencies
+        uses: "./.github/actions/install_node_dependencies"
+
+      - name: Build React application
+        run: |-
+          node -v
+          make build
+
+      - name: Install Kedro-Viz development package
+        run: |
+          pip install -e package
```

Comment:
I hope if we want to release the workflow, we will install it from pypi and this is just for testing ?

Reply:
We do this until next kedro-viz releease. Since current pypi won't have the changes in this branch. I am thinking 
- we push this branch to main and then to demo (we don't do any release) 
- once we do the release and know that the latest code is on Pypi - we test it with public-kedro-viz action and update this. Wdyt?